### PR TITLE
TransformControls.js: Added stop() and cancel() methods

### LINKED ISSUE #21655: Adding cancel method to TransformControls helper
**Is your feature request related to a problem? Please describe.**

If you use the `TransformControls` helper in your scenes, like in the [example](https://threejs.org/examples/?q=Transform#misc_controls_transform), it's not possible to easily cancel and undo a current transform action.

**Describe the solution you'd like**

It would be nice to have a `cancel` method on the helper that would restore the object to its state before the current transform action began.

This would require a temporary copy of the object (or object's rotation, position and scale) made on mousedown.
This copy can be discarded on mouseup. But if someone would call the `cancel` method while dragging with the mouse, it would cancel/stop the transform action and restore the object to its former state.

**Describe alternatives you've considered**

***Option 1: `.enabled = false`***
There is the `.enabled = false` to stop the transformation action, BUT it doesn't really cancel the transformation, it just freezes it temporarily.
Like in the following case with the [example](https://threejs.org/examples/?q=Transform#misc_controls_transform):
1) start dragging one of the translation axes
2) press `space` to disable the controls (the object is now frozen in the last place it was dragged to)
3) release your mouse button (should clear all the event listeners and prepare for a new transform, but doesn't because of [this check](https://github.com/mrdoob/three.js/blob/e48fc94dfeaecfcbfa977ba67549e6108b370cbf/examples/js/controls/TransformControls.js#L647))
4) press `space` again to re-enable the controls to make a new edit...
Oops.. now the original transform action is activated again or, more accurately, is still active).

***Option 2: Workaround***
Because the internal functions are not accessible from the outside, I had to make a workaround like this:
1) A `mouseDown` listener that
    - stores the initial object's position, rotation and scale
    - adds a `keydown` event listener to catch a `Escape` keypress (to cancel and undo the transform)
2) The keydown listener
    - restores the object's state
    - calls `.detach()` on the controls instance (to ensure the object is not changed anymore until mouseup)
    - adds a `pointerup` listener to the `.domElement.ownerDocument`
3) The pointerup listener
    - re-attaches the object to the transform controls
    - and removes itself as listener.
4) A `mouseUp` listener that removes the above keydown listener.

**Additional context**

In addition, it might be useful to put the functionality in two separate methods `stop` and `cancel`, where `stop` would just trigger the mouseup event and `cancel` would restore the object's state and then call `stop`.

### PR DESCRIPTION
**Pull request by vHeemstra**
Related issue: #21655

**Description**

There was no (easy) way to cancel/abort a transform action. The pull request adds that functionality.
- `stop()` does the same as a regular mouse up event.
- `cancel()` restores the current object to its state when the dragging/transformation started and fires `stop()`. The optional `continueTransform` boolean parameter can prevent the firing of `stop()` to let the user continue with the transform action.

Resolves #21655

### CODE REVIEW

**Review by c-z-chen — APPROVED** [review]

**Comment by mrdoob** [comment]
@arodic Is this something that has been solved in the new `TransformControls`?

**Comment by asfg84** [comment]
When will this feature be available? I would also like to use it in my applications...

**Comment by Mugen87** [comment]
I vote to add both methods. The use case described in #21655 is comprehensible.

A few things have changed in `TransformControls` though so a rebase is required. If the OP does not have the time, I can also clear things up.

**Comment by vHeemstra** [comment]
I will take a look at the current code base and will make the methods for that.

**Comment by vHeemstra** [comment]
I made the method in the current code base. Should I open a new pull request?

**Comment by vHeemstra** [comment]
I made a new pull request, so I will close this one.
